specs/experimental/op-stack-manager.md

```diff
@@ -0,0 +1,259 @@
+[superchain registry]: https://github.com/ethereum-optimism/superchain-registry
+[ethereum-lists/chains]: https://github.com/ethereum-lists/chains
+[Batch Inbox]: ../protocol/configurability.md#consensus-parameters
+
+The OP Stack Manager is a contract that deploys the L1 contracts for an OP Stack chain in a single
+transaction. It provides a minimal set of user-configurable parameters to ensure that the resulting
+chain meets the [standard configuration] requirements.
+
+The version deployed is always a governance-approved contract release. The set
+of governance approved [contract releases] can be found on the
+[Optimism Monorepo releases] page, and is the set of releases named
+`op-contracts/vX.Y.Z`.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Deployment](#deployment)
+- [Interface](#interface)
+  - [`deploy`](#deploy)
+  - [Getter Methods](#getter-methods)
+- [Implementation](#implementation)
+  - [Batch Inbox Address](#batch-inbox-address)
+  - [Contract Deployments](#contract-deployments)
+- [Security Considerations](#security-considerations)
+  - [Chain ID Source of Truth](#chain-id-source-of-truth)
+  - [Chain ID Frontrunning](#chain-id-frontrunning)
+  - [Chain ID Value](#chain-id-value)
+  - [Proxy Admin Owner](#proxy-admin-owner)
+  - [Upgradeability (ABI Changes)](#upgradeability-abi-changes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Deployment
+
+The OP Stack Manager is deployed at `0xTODO`. It can be deployed as follows:
+
+TODO.
+
+## Interface
+
+Version 1.0.0 of the OP Stack Manager deploys the [`op-contracts/v1.3.0`]
+contracts release.
+
+### `deploy`
+
+The `deploy` method is the only non-view method in the contract. It is used to
+deploy the full set of L1 contracts required to setup a new OP Stack chain that
+complies with the [standard configuration]. It has the following interface:
+
+```solidity
+function deploy(
+  uint64 l2ChainId,
+  address proxyAdminOwner,
+  SystemConfigInputs memory systemConfigInputs,
+  L2OutputOracleInputs memory l2OutputOracleInputs
+) external returns (SystemConfig systemConfig);
+```
+
+where:
+
+- `l2ChainId` is the chain ID of the OP Stack chain.
+- `proxyAdminOwner` is the address that will own the `ProxyAdmin` contract that will be deployed.
+This corresponds to the L1 ProxyAdmin owner and L2 ProxyAdmin owner [admin roles].
+- `systemConfigInputs` is a struct defining configuration of the `SystemConfig` inputs.
+- `l2OutputOracleInputs` is a struct defining configuration of the `L2OutputOracle` inputs.
+
+The `l2ChainId` has the following restrictions:
+
+- It is limited to a `uint64`. This contradicts the [interop specification],
+which allows chain ID to be a `uint256`. This contradiction exists because it is
+currently unspecified how to unify a 256-bit chain ID with the existing batch inbox
```

Comment:
I think that we can remove mention to #94 here, this isn't super relevant to the spec. I think that we should just decided on a simple enough solution for #94, its impossible to preserve all of the data so we could just truncate

Reply:
Removed, so resolving this in favor of https://github.com/ethereum-optimism/specs/pull/236#discussion_r1648230928